doc/reference/orm.rst

```diff
@@ -445,503 +600,173 @@ stored::
-
-        :type: :class:`~odoo.fields.Many2one`
-
-    .. attribute:: parent_path
-
-        used to store an index of the tree structure when :attr:`~._parent_store`
-        is set to True - must be declared with ``index=True`` for proper operation.
-
-        :type: :class:`~odoo.fields.Char`
-
-
-.. _reference/orm/decorators:
+.. _reference/api/decorators:
 
 Method decorators
 =================
 
 .. automodule:: odoo.api
-    :members: model, depends, constrains, onchange, returns,
-
-.. _reference/orm/fields:
-
-Fields
-======
-
-.. _reference/orm/fields/basic:
-
-Basic fields
-------------
-
-.. autodoc documents descriptors as attributes, even for the *definition* of
-   descriptors. As a result automodule:: odoo.fields lists all the field
-   classes as attributes without providing inheritance info or methods (though
-   we don't document methods as they're not useful for "external" devs)
-   (because we don't support pluggable field types) (or do we?)
-
-.. autoclass:: odoo.fields.Field
-
-.. autoclass:: odoo.fields.Char
-    :show-inheritance:
-
-.. autoclass:: odoo.fields.Boolean
-    :show-inheritance:
+    :members: depends, depends_context, constrains, onchange, returns
 
-.. autoclass:: odoo.fields.Integer
-    :show-inheritance:
+.. .. currentmodule:: odoo.api
 
-.. autoclass:: odoo.fields.Float
-    :show-inheritance:
+.. .. autodata:: model
+.. .. autodata:: depends
+.. .. autodata:: constrains
+.. .. autodata:: onchange
+.. .. autodata:: returns
 
-.. autoclass:: odoo.fields.Text
-    :show-inheritance:
+.. todo:: With sphinx 2.0 : autodecorator
 
-.. autoclass:: odoo.fields.Selection
-    :show-inheritance:
+.. _reference/api/onchange:
 
-.. autoclass:: odoo.fields.Html
-    :show-inheritance:
-
-.. _reference/orm/fields/date_datetime:
-
-Date and Datetime fields
-------------------------
-
-Dates and Datetimes are very important fields in any kind of business
-application, they are heavily used in many popular Odoo applications such as
-logistics or accounting and their misuse can create invisible yet painful
-bugs, this excerpt aims to provide Odoo developers with the knowledge required
-to avoid misusing these fields.
-
-When assigning a value to a Date/Datetime field, the following options are valid:
-    * A string in the proper server format **(YYYY-MM-DD)** for Date fields,
-      **(YYYY-MM-DD HH:MM:SS)** for Datetime fields.
-    * A `date` or `datetime` object.
-    * `False` or `None`.
-
-If not sure of the type of the value being assigned to a Date/Datetime object,
-the best course of action is to pass the value to
-:func:`~odoo.fields.Date.to_date` or :func:`~odoo.fields.Datetime.to_datetime`
-which will attempt to convert the value to a date or datetime object
-respectively, which can then be assigned to the field in question.
-
-.. admonition:: Example
-
-    To parse date/datetimes coming from external sources::
-
-        fields.Date.to_date(self._context.get('date_from'))
+onchange: updating UI on the fly
```

Comment:
Why does onchange has its own subsection in Method decorators while other decorators don't ?
Also, it is already introduced in the parent's section. It's weird to re-explain the decorator because the use-case is different.

Reply:
After discussion,
* Merge this with the docstring and remove this section.
* In the docstring(**s**), explain the purpose of the decorator before the example code block.